pkg/store/bucket.go

```diff
@@ -237,6 +241,19 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
 		Buckets: []float64{0.001, 0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20, 30, 60, 90, 120},
 	})
 
+	m.blockSyncs = promauto.With(reg).NewCounter(prometheus.CounterOpts{
```

Comment:
[non-blocking] I think it's more handy when we have a count of the total number of attempts (eg. `thanos_bucket_store_block_syncs_total`), regardless they succeeded or not, and then we can failures as you did in `blockSyncFailures`.

This way you can compute the failure rate as `thanos_bucket_store_block_syncs_failures_total  / thanos_bucket_store_block_syncs_total` instead of having to do `thanos_bucket_store_block_syncs_failures_total / (thanos_bucket_store_block_syncs_failures_total + thanos_bucket_store_block_syncs_successful_total)`

Reply:
I agree with @pracucci. WDYT @zecke? Would you mind making the change? Then we can merge.